nemo_curator/scripts/semdedup/clustering.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+from datetime import datetime
+
+os.environ["DASK_DATAFRAME__QUERY_PLANNING"] = "False"
+import dask_cudf
+
+from nemo_curator.datasets import DocumentDataset
+from nemo_curator.log import create_logger
+from nemo_curator.modules.config import SemDedupConfig
+from nemo_curator.modules.semantic_dedup import ClusteringModel
+from nemo_curator.utils.distributed_utils import get_client
+from nemo_curator.utils.script_utils import ArgumentHelper
+
+if __name__ == "__main__":
+    # Configure command line arguments
+    semdedup_config = SemDedupConfig.from_yaml("config.yaml")
```

Comment:
Same comments as with `compute_embeddings.py`, move this to argparse, use `expand_outdir_and_mkdir`, etc.

Reply:
Resolved by https://github.com/NVIDIA/NeMo-Curator/pull/130/commits/a304629f572c0bb202bd14efa5843b18bb7ca388